src/feditest/nodedrivers/mastodon/__init__.py

```diff
@@ -0,0 +1,373 @@
+        self,
+        candidate_member_uri: str,
+        collection_uri: str,
+        spec_level: SpecLevel | None = None,
+        interop_level: InteropLevel | None= None
+    ):
+        collection = AnyObject(collection_uri).as_collection()
+        if not collection.contains_item_with_id(candidate_member_uri):
+            raise AssertionFailure(
+                spec_level or SpecLevel.UNSPECIFIED,
+                interop_level or InteropLevel.UNKNOWN,
+                f"{candidate_member_uri} not in {collection_uri}")
+
+
+    # Python 3.12 @override
+    def assert_not_member_of_collection_at(
+        self,
+        candidate_member_uri: str,
+        collection_uri: str,
+        spec_level: SpecLevel | None = None,
+        interop_level: InteropLevel | None= None
+    ):
+        collection = AnyObject(collection_uri).as_collection()
+        if collection.contains_item_with_id(candidate_member_uri):
+            raise AssertionFailure(
+                spec_level or SpecLevel.UNSPECIFIED,
+                interop_level or InteropLevel.UNKNOWN,
+                f"{candidate_member_uri} must not be in {collection_uri}")
+
+
+# From WebFingerServer
+    # Python 3.12 @override
+    def obtain_account_identifier(self, rolename: str | None = None) -> str:
+        trace(f'obtain_account_identifier for role {rolename}')
+        user = self._local_users_by_role.get(rolename)
+        if not user:
+            user = self._provision_new_user()
+            self._local_users_by_role[rolename] = user
+        return f'acct:{ user._userid }@{ self.parameter( "hostname" )}'
+
+
+    # Python 3.12 @override
+    def obtain_non_existing_account_identifier(self, rolename: str | None = None ) -> str:
+        trace(f'obtain_non_existing_account_identifier for role {rolename}')
+        userid = self._non_existing_userids_by_role[rolename]
+        if not userid:
+            userid = self._create_non_existing_user()
+            self._non_existing_userids_by_role[rolename] = userid
+        return userid
+
+
+    # Not implemented
+    # def obtain_account_identifier_requiring_percent_encoding(self, nickname: str | None = None) -> str:
+    # def override_webfinger_response(self, client_operation: Callable[[],Any], overridden_json_response: Any):
+
+# From WebServer
+
+    # Not implemented:
+    # def transaction(self, code: Callable[[],None]) -> WebServerLog:
+    # def override_http_response(self, client_operation: Callable[[],Any], overridden_json_response: Any):
+
+# Internal implementation helpers
+
+    def _userid_to_actor_uri(self, userid: str) -> str:
+        """
+        The algorithm by which this application maps userids to ActivityPub actor URIs.
+        Apparently this is different between Mastodon and other implementations, such as WordPress,
+        so this might be overridden
+
+        see also: _actor_uri_to_userid()
+        """
+        return f'https://{ self.parameter("hostname") }/users/{ userid }'
```

Comment:
It may not need to be overridden if you use the actor URI provided by the Mastodon-compatible instance in the account information. The original Mastodon API node code used that approach for obtaining the actor URI. It should work for any Mastodon-compatible API.